include/jwt-cpp/jwt.h

```diff
@@ -107,58 +307,165 @@ namespace jwt {
 	 * you maybe need to extract the modulus and exponent of an RSA Public Key.
 	 */ 
 	namespace helper {
+		/**
+		 * \brief Extract the public key of a pem certificate
+		 * 
+		 * \param certstr	String containing the certificate encoded as pem
+		 * \param pw		Password used to decrypt certificate (leave empty if not encrypted)
+		 * \param ec		error_code for error_detection (gets cleared if no error occures)
+		 */
 		inline
-		std::string extract_pubkey_from_cert(const std::string& certstr, const std::string& pw = "") {
+		std::string extract_pubkey_from_cert(const std::string& certstr, const std::string& pw, std::error_code& ec) {
+			ec.clear();
 #if OPENSSL_VERSION_NUMBER <= 0x10100003L
 			std::unique_ptr<BIO, decltype(&BIO_free_all)> certbio(BIO_new_mem_buf(const_cast<char*>(certstr.data()), certstr.size()), BIO_free_all);
 #else
 			std::unique_ptr<BIO, decltype(&BIO_free_all)> certbio(BIO_new_mem_buf(certstr.data(), static_cast<int>(certstr.size())), BIO_free_all);
 #endif
 			std::unique_ptr<BIO, decltype(&BIO_free_all)> keybio(BIO_new(BIO_s_mem()), BIO_free_all);
+			if(!certbio || !keybio) {
+				ec = error::rsa_error::create_mem_bio_failed;
+				return "";
```

Comment:
```suggestion
				return {};
```

better optimization https://godbolt.org/z/93Phdq

Reply:
This could be applied throughout, maybe a separate PR?